test/sequential/test-tls-psk-client.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const tls = require('tls');
+const spawn = require('child_process').spawn;
+
+const CIPHERS = 'PSK+HIGH';
+const KEY = 'd731ef57be09e5204f0b205b60627028';
+const IDENTITY = 'Client_identity';  // Hardcoded by `openssl s_server`
+
+const server = spawn(common.opensslCli, [
+  's_server',
+  '-accept', common.PORT,
+  '-cipher', CIPHERS,
+  '-psk', KEY,
+  '-psk_hint', IDENTITY,
+  '-nocert'
+]);
+
+let state = 'WAIT-ACCEPT';
+
+let serverStdoutBuffer = '';
+server.stdout.setEncoding('utf8');
+server.stdout.on('data', (s) => {
+  serverStdoutBuffer += s;
+  switch (state) {
+    case 'WAIT-ACCEPT':
+      if (/ACCEPT/g.test(serverStdoutBuffer)) {
+        // Give s_server a second to start up.
+        setTimeout(startClient, 1000);
```

Comment:
This worries me. spawn latency varies by platform, is this robust on arm? Is it possible to poll infinitely for the server port to open at the TCP level before starting the client, so that the test has until the framework terminates in on timeout to wait for the server?

Reply:
I had hoped to remove this along with the fix for Windows of this test and I tried to do that once again.
Though no fix yet, but I discovered that OpenSSL server only works with either `{ detached: true, shell: true }` or `{ stdio: 'inherit', shell: true }` by which it looks like they use the actual shell in a server and for some reason simple `'pipe'`'s doesn't work.

As for the delay I'd want to first try to remove it because after we've got `ACCEPT` from the server I'd assume that it had already opened a port for new connections.

Otherwise I think I'd try the 'poll' tactic.

Reply:
Finally managed to fix this test. PTAL.